src/annotator/highlighter.js

```diff
@@ -150,19 +144,48 @@ function drawHighlightsAbovePdfCanvas(highlightEl) {
  * @typedef {HTMLElement & HighlightProps} HighlightElement
  */
 
+/**
+ * Return the complete text nodes inside `range`.
+ *
+ * If a range starts or ends part-way through a text node, the nodes are split
+ * and the part inside the range is returned.
+ *
+ * @param {Range} range
+ * @return {Text[]}
+ */
+function wholeTextNodesInRange(range) {
+  const textNodes = [];
+
+  forEachNodeInRange(range, node => {
+    if (node.nodeType !== Node.TEXT_NODE) {
+      return;
+    }
+    let text = /** @type {Text} */ (node);
+
+    if (text === range.startContainer && range.startOffset > 0) {
+      text = text.splitText(range.startOffset);
```

Comment:
This line appears to be a test miss. Can you rectify that?

Reply:
Indeed! - I've added a bunch of extra tests and caught a bug in the process.